src/prettytable/prettytable.py

```diff
@@ -1454,6 +1454,7 @@ def _get_options(self, kwargs: Mapping[str, Any]) -> OptionsType:
 
     def set_style(self, style: TableStyle) -> None:
         self._style = style
+        self._set_default_style()
```

Comment:
Little nit, so when we read the `if` statement, the `style` var is still in our head :)

```suggestion
        self._set_default_style()
        self._style = style
```